pandas-stubs/_typing.pyi

```diff
@@ -155,6 +156,9 @@ AxisType: TypeAlias = Literal["columns", "index", 0, 1]
 DtypeNp = TypeVar("DtypeNp", bound=np.dtype[np.generic])
 KeysArgType: TypeAlias = Any
 ListLike = TypeVar("ListLike", Sequence, np.ndarray, "Series", "Index")
+ListLikeExceptSeriesAndStr = TypeVar(
+    "ListLikeExceptSeriesAndStr", MutableSequence, np.ndarray, tuple, "Index"
+)
```

Comment:
I agree, the name is awful, but do we have a better way to exclude `str` from list-likes?

Reply:
I don't know of any